src/registrations.js

```diff
@@ -26,23 +38,44 @@ export function singleton(keyOrRegisterInChild?: any, registerInChild: boolean =
 }
 
 /**
+ * Decorator: Specifies to register the decorated item as a configuration class.
+ * @param child Should a child container be created?
+ * @returns {any}
+ */
+export function config(child?: boolean): any {
+  return registration(new ConfigurationRegistration(child));
+}
+
+/**
 * Customizes how a particular function is resolved by the Container.
 */
-export interface Registration {
+export class Registration {
```

Comment:
The reason for it is to enable some common functionality. The alternative would be copy paste common code, or (a better alternative) move to a standalone function that you would have to _remember_ to call along the every time you need the functionality. I'm fine with that but it would seem counterintuitive in this case.